src/cpp/security/cryptography/AESGCMGMAC_Transform.cpp

```diff
@@ -926,6 +924,31 @@ bool AESGCMGMAC_Transform::preprocess_secure_submsg(
                     }
                 }   //For each Reader2WriterKeyMaterial in the local datareader
             } //For each datareader present in the local participant
+
+            // Discovery: local datareader not found, look for remote datareader only when is_key_id_zero is true
+            if (is_key_id_zero)
+            {
+                for (std::vector<DatareaderCryptoHandle*>::iterator itt = remote_participant->Readers.begin();
```

Comment:
I would put the code in this loop on a separate method and use it here and above. Something like `lookup_reader`.

I would also change the loop into a ranged for.

Reply:
Done